packages/paste-core/components/sidebar/__tests__/sidebarNavigation.spec.tsx

```diff
@@ -0,0 +1,315 @@
+                  <SidebarBetaBadge as="span">Beta</SidebarBetaBadge>
+                </Box>
+                <SidebarNavigationDisclosureContent>
+                  <SidebarNavigationItem>Navigation Item</SidebarNavigationItem>
+                  <SidebarNavigationItem selected>Navigation Item</SidebarNavigationItem>
+                  <SidebarNavigationItem>Navigation Item</SidebarNavigationItem>
+                </SidebarNavigationDisclosureContent>
+              </SidebarNavigationDisclosure>
+              <SidebarNavigationItem onClick={onClick}>Navigation Item</SidebarNavigationItem>
+              <SidebarNavigationItem as="a" href="https://google.com" selected>
+                Navigation Item
+              </SidebarNavigationItem>
+              <SidebarNavigationItem>Navigation Item</SidebarNavigationItem>
+            </SidebarNavigationDisclosureContent>
+          </SidebarNavigationDisclosure>
+        </SidebarNavigation>
+        <SidebarCollapseButtonWrapper>
+          <SidebarCollapseButton onClick={onClick} i18nCollapseLabel="Close sidebar" i18nExpandLabel="Open sidebar" />
+        </SidebarCollapseButtonWrapper>
+      </Sidebar>
+
+      {/* Must wrap content area */}
+      <SidebarPushContentWrapper collapsed={collapsed} variant="compact">
+        <Button variant="primary" onClick={onClick}>
+          Toggle Push Sidebar
+        </Button>
+      </SidebarPushContentWrapper>
+    </Theme.Provider>
+  );
+  /* eslint-enable react/jsx-max-depth */
+};
+
+describe('SidebarNavigation', () => {
+  it('renders collapsed SidebarNavigationItems correctly', () => {
+    const onClick: jest.Mock = jest.fn(() => {});
+
+    render(<MockPushSidebarWithNavigation collapsed onClick={onClick} />);
+    const wrapper = screen.getByTestId('nav-wrapper');
+    const buttonIcon = screen.getByTestId('nav-item-button');
+    const anchorIcon = screen.getByTestId('nav-item-anchor');
+    const disclosure = screen.getByTestId('nav-item-disclosure');
+
+    /*
+     * Check that the sidebar has the correct number of children, including the disclosure
+     * We check this so that in the future we don't remove the hidden disclosure
+     * on collapsed, because if we do that then we need to manage the state of the
+     * disclosure in the collapsed state
+     */
+    expect(wrapper.children).toHaveLength(3);
+
+    // Button works
+    expect(buttonIcon).toBeInTheDocument();
+    fireEvent.click(buttonIcon);
+    expect(onClick).toBeCalledTimes(1);
+    // N.B: 'getByText' fails when element is missing so we use 'queryByText' here
+    const buttonByText = screen.queryByText('This item closes the sidebar');
+    expect(buttonByText).not.toBeInTheDocument();
+
+    // Anchor works
+    expect(anchorIcon).toHaveAttribute('href');
+
+    // Disclosure works
+    expect(disclosure).toBeInTheDocument();
+    expect(disclosure).toHaveStyleRule('display', 'none');
+  });
+
+  it('renders expanded SidebarNavigationItems correctly', () => {
```

Comment:
question: are there any aria-attributes you want to be ensuring work correctly with the user interaction?